tools/lint_diff.py

```diff
@@ -52,10 +52,17 @@ def find_branch_point(branch):
         'Failed to find a common ancestor in the last 1000 commits')
 
 
-def find_diff(sha):
+def find_diff(sha, files=None):
     """Find the diff since the given sha."""
+    if files:
+        for file_or_dir in files:
+            msg = f"{file_or_dir} doesn't exist. Please provide a valid path."
+            assert os.path.exists(file_or_dir), msg
+    else:
+        files = '*.py'
+
     res = subprocess.run(
-        ['git', 'diff', '--unified=0', sha, '--', '*.py'],
+        ['git', 'diff', '--unified=0', sha, '--', *files],
```

Comment:
Wait... so the only thing that was missing was the `*`. This suggests to me that whatever command was being sent to `flake8` produced no output. Is that correct? So there _are_ `flake8` commands for which `flake8` does not actually run correctly and produces no output... and `flake8` also produces no output when it runs correctly and finds no issues? There was at least some truth to the original text in discussion [here](https://github.com/scipy/scipy/pull/14507/files#r680460592)?

Reply:
> This suggests to me that whatever command was being sent to flake8 produced no output. Is that correct?

The issue has nothing to do with flake8. It was a problem with how diff was being calculated earlier. See my example below. Btw, I just realized that there was another issue with `*'*.py'` (although it did not really affect the output of the program) which is now fixed.

**Explaining the fix:**
When you pass multiple files/dirs, they were being parsed as a list of strings. And later I was combining the list into one full string using `files=' '.join(files)`. But this did not work (see case 2 in example below) with the `subprocess.run` command within Python.

Instead, now I use the * operator to unpack the list and directly feed it into the command which works as expected because the arguments are passed in the case 1 mentioned below.

**Example:**
You can make small changes to somewhere in `ndimage` and `test_hypotests.py` stats and then try the following:

If you just do the same directly in the terminal, like so:
CASE 1:
`git diff --unified=0 -- scipy/stats/tests/test_hypotests.py scipy/ndimage`
The above would work but passing the files/dirs within the strings won't work (below)
CASE 2:
``git diff --unified=0 -- 'scipy/stats/tests/test_hypotests.py scipy/ndimage'``

Reply:
So in summary this had nothing to do with flake8 and the removal [here in your comment](https://github.com/scipy/scipy/pull/14507/files#r680460592) should be done. Sorry for the confusion.